src/java.base/share/classes/jdk/internal/jimage/ImageReader.java

```diff
@@ -273,448 +272,413 @@ public void close(ImageReader image) throws IOException {
             }
         }
 
-        void addOpener(ImageReader reader) {
-            synchronized (OPEN_FILES) {
-                openers.add(reader);
-            }
-        }
-
-        boolean removeOpener(ImageReader reader) {
-            synchronized (OPEN_FILES) {
-                return openers.remove(reader);
-            }
-        }
-
-        // directory management interface
-        Directory getRootDirectory() {
-            return buildRootDirectory();
-        }
-
         /**
-         * Lazily build a node from a name.
+         * Returns a node with the given name, or null if no resource or directory of
+         * that name exists.
+         *
+         * <p>This is the only public API by which anything outside this class can access
+         * {@code Node} instances either directly, or by resolving symbolic links.
+         *
+         * <p>Note also that there is no reentrant calling back to this method from within
+         * the node handling code.
```

Comment:
This paragraph and the previous one feels like too much detail and I'm not sure they are necessary.